ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/models/Parameter.java

```diff
@@ -61,4 +63,19 @@ public String toString() {
         }
         return parameterType + parameterPointerType + " " + name;
     }
+
+    public boolean equals(final Object obj) {
+        if (!(obj instanceof Parameter)) {
+            return false;
+        }
+        return getName().equals(((Parameter) obj).getName())
```

Comment:
You should create a new variable for the object you are comparing against so that you only have to cast it once.